### PR TITLE
Support latest version of web3

### LINKED ISSUE #8: Need to support the latest web3
I spent several hours today trying to integrate this functionality into an existing project and had a frustrating time figuring out what was wrong. 

In the other project I already have Ganache-CLI configured and running with Ethereum-Bridge connected to it. I can deploy my own contracts, but now I want some of this provableAPI juice. 

When I cloned this project, did `npm install`, ran `truffle console` and `test` the tests all passed. So I copied the contracts and test files over to my project where a few tests passed, one failed and the final one hung.

I went back and forth between projects trying to figure out what was different, and eventually got the tests running in my project by rewriting somewhat. 

Eventually, I stumbled upon the fact that you have a fixed version of web3 in your package.json. I have `"web3": "^1.0.0-beta.55"` and you have `"web3": "1.0.0-beta.36"`. I upgraded your `web-3` and suddenly I saw the same errors I was seeing in my own project.

This is a fairly old version of `web-3`, yet I notice that there was a commit to your project just six days ago. If you don't use the ^ in your dependency declaration, you're won't catch problems like this and stay up to date. 

**Tests failing with `"web3": "^1.0.0-beta.55"`**
Screen Shot 2019-07-17 at 4 50 44 PM

### PR DESCRIPTION
**Pull request by cliffhall**
This PR fixes #8 

* In contracts/EthPrice.sol
  - refactor/renamed fetchEthPriveViaProvable to fetchEthPriceViaProvable (mispelling)

* Refactor/moved test/util.js to test/util/Events.js

* Added test/util/Exceptions.js for absracting the handling of revert or other types of transaction failures

* In test/eth-price-tests.js,
  - import waitForEvent from util/Events
  - import exceptions from util/Exceptions
  - in test 'Should set ETH price correctly in contract', convert return of call with toNumber()
  - in test 'Should revert on second query attempt due to lack of funds',
    - rewrite try/catch to use exceptions.catchRevert
    - refactored fetchEthPriveViaProvable to fetchEthPriceViaProvable (mispelling)
  - In test 'Should succeed on a second query attempt when sending funds',
    - refactored fetchEthPriveViaProvable to fetchEthPriceViaProvable (mispelling)
    - dont' await call to fetchEthPriceViaProvable
    - instead, await waitForEvent to get the event data then assert on its description

 * In package.json,
   - changed ethereum-bridge version selector to "^0.6.2"
   - changed web3 version selector to "^1.0.0-beta.55"

**Tests passing with `"web3": "^1.0.0-beta.55"`**
Screen Shot 2019-07-17 at 5 25 24 PM

### CODE REVIEW

**Comment by cliffhall** [comment]
Note that there are minor port config differences on my personal setup, but I did not modify this project for those things.

I have `Ganache-CLI` running on 7545, so my `truffle.js` has this:

```
        development: {
            host: "127.0.0.1",
            port: 7545,
            network_id: "*"
        },
 ```
Also, I'm starting `Ethereum-Bridge` with: 
`"provable:start": "npx ethereum-bridge --dev -H localhost:7545 -a 1"`

So in my `eth-price-tests.js`, I connect the websocket to my Ganache instance with:
`const web3 = new Web3(new Web3.providers.WebsocketProvider('ws://localhost:7545'));`

**Comment by D-Nice** [comment]
This very well put together PR is quite appreciated @cliffhall

To give some explanation as to why the web3 dep was version locked, it is due to the fact we noticed various issues in later versions at that time, which is to be expected with a beta build and a general effect of working in a bleeding edge ecosystem as you likely very well know :). We had found 36 at the time to be the most stable and therefore locked the dependency down there.

Of course, if things have improved since, it makes sense to support the latest. I'll ask our ETH lead to have a look.

**Review by gskapka — APPROVED** [review]
Thanks @cliffhall - your patch applies cleanly and works on my end. Will probably switch to a Zeppelin solution in future for the revert checks since we use and support their work elsewhere, but that'll be for another internal PR. 

Am also going to leave in the stylistic/linting differences for now rather than burden you with changing them since this repo is likely to undergo another PR cycle before the upcoming Truffle conference.

Tl;dr:__ Patch works cleanly, happy to merge & thank you for your help!

**Comment by cliffhall** [comment]
Happy to help!
